poetry.lock

```diff
@@ -1007,9 +1011,40 @@ python-versions = ">=3.7"
 [package.extras]
 idna2008 = ["idna"]
 
+[[package]]
+name = "rich"
+version = "12.5.1"
+description = "Render rich text, tables, progress bars, syntax highlighting, markdown and more to the terminal"
+category = "dev"
+optional = false
+python-versions = ">=3.6.3,<4.0.0"
+
+[package.dependencies]
+commonmark = ">=0.9.0,<0.10.0"
+pygments = ">=2.6.0,<3.0.0"
+typing-extensions = {version = ">=4.0.0,<5.0", markers = "python_version < \"3.9\""}
+
+[package.extras]
+jupyter = ["ipywidgets (>=7.5.1,<8.0.0)"]
+
+[[package]]
+name = "rust-python-jaeger-reporter"
```

Comment:
How do I add only `rust-python-jaeger-reporter` without changing the rest of the `poetry.lock` file?

```
poetry add git+https://github.com/erikjohnston/rust-jaeger-python-client.git
```

Reply:
Maybe `poetry lock --no-update` but that still seems to change things

Reply:
`poetry lock --no-update` is what you want. You might need to update to poetry [1.1.14](https://github.com/python-poetry/poetry/releases/tag/1.1.14) and [clear caches](https://matrix-org.github.io/synapse/develop/development/dependencies.html#clear-caches-poetry-cache-clear---all-pypi) to workaround this.

Reply:
Thanks @DMRobertson, it seemed to change less stuff this time after clearing the cache but still looks dubious, https://github.com/matrix-org/synapse/pull/13399/commits/b732dabe3d0740462f06aa780a962c881ea8bcbd
```
$ poetry --version
Poetry version 1.1.14
$ poetry cache list
pypi
$ poetry cache clear --all pypi
Delete 605 entries? (yes/no) [no] yes
$ poetry lock --no-update
Resolving dependencies... (125.3s)

Writing lock file

# Commited the result
```

Reply:
The closer I look at it, maybe it's just re-ordering some stuff which is fine 🤷

Reply:
I saw this earlier today on another branch. I don't have a good explanation either :(